ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import { Flex } from '@patternfly/react-core';
+
+import { SelectOption } from '@patternfly/react-core/deprecated';
+import { CompoundSearchFilterConfig } from '../types';
+
+import EntitySelector from './EntitySelector';
+
+export type CompoundSearchFilterProps = {
+    config: CompoundSearchFilterConfig;
+};
+
+function CompoundSearchFilter({ config }: CompoundSearchFilterProps) {
+    const entities = Object.keys(config);
+
+    const [selectedEntity, setSelectedEntity] = useState(() => {
+        return entities[0];
+    });
+
+    return (
+        <Flex>
+            {entities.length > 0 && (
+                <EntitySelector
+                    value={selectedEntity}
+                    onChange={(value) => setSelectedEntity(value)}
+                >
+                    {entities.map((entity) => {
+                        return (
+                            <SelectOption key={entity} value={entity}>
+                                {entity}
+                            </SelectOption>
+                        );
+                    })}
```

Comment:
What is the benefit of creating these options here, versus passing `entities` to the `<EntitySelector>` and letting it handle creating the options?

Reply:
Good question. I was initially going to encapsulate the logic into `EntitySelector`, but it didn't seem like there was much to encapsulate. I'm rethinking a few things as I'm working on the `AttributeSelector`. I'll make a few changes in that PR